site/components/pages/Retirements/SingleRetirement/ViewPledgeButton/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { ButtonPrimary } from "@klimadao/lib/components";
+import { t } from "@lingui/macro";
+import { Pledge } from "components/pages/Pledge/types";
+import { FC } from "react";
+
+type ViewPledgeButtonProps = {
+  pledge: Pledge | null;
+};
+
+/** Logic for hiding or showing a link to the users pledge depending on a valid pledge */
+export const ViewPledgeButton: FC<ViewPledgeButtonProps> = (props) => {
+  const hasPledge = props.pledge !== null;
+  /** @todo we need a better way of accessing the pledge url. This will break on any url change */
+  const pledgeUrl = hasPledge
+    ? `https://www.klimadao.finance/pledge/${props.pledge?.ownerAddress}`
```

Comment:
Please use the `urls.pledges` as aeterno described

Reply:
Absolutely, done 👍